tests/test_marquez_dag.py

```diff
@@ -455,9 +444,9 @@ def test_marquez_dag_with_extractor(mock_get_or_create_marquez_client,
         log.info(call)
 
     assert [name for name, args, kwargs in mock_marquez_client.mock_calls] == [
-       'create_dataset',  # we would expect only the output to be updated
-       'create_dataset',
-       'create_job',
-       'create_job',  # we would expect only one call to update the job
-       'mark_job_run_as_completed'
+        'create_dataset',  # we would expect only the output to be updated
+        'create_dataset',
+        'create_job',
+        'mark_job_run_as_started',  # Emit when the task instance has started
+        'mark_job_run_as_completed'
```

Comment:
I think we need two versions of `test_marquez_dag_with_extractor`. One that tests when the extractor extracts the lineage in `extract` and one when it does so in `extract_on_complete`. that's how you'll cover both cases.
That should deal with most of the codecov comments in your PR.